### PR TITLE
UI: Update to use Grafana's design system

### LINKED ISSUE #25: Replace current UI elements with ones from Grafana Design System
Look the Grafana Design System docs [here](https://developers.grafana.com/ui/latest/index.html?path=/story/docs-overview-intro--page).

### LINKED ISSUE #33: Share button repositioning

### LINKED ISSUE #34: Move the Run button to be central and the main action on the playgroud
Move the Run button to be central and the main action on the playground
The button should have both the symbol and the "Run" label (as https://www.w3schools.com/tryit/tryit.asp?filename=tryhtml_hello)
The button should follows the design system guidelines for the main action button in a page. Ik they don't exists, work with our designer to obtain them.
The two fields before the button are part of the group and should be moved together with the Run button. 
It would be useful to have a label before each of them. Like "Select operation" and "Select version". Check the guidelines for the shape of the boxes, rules on single selection lists, and if the two should be stacked on top of each other or on the same line. 
The Version should only be enabled to select when appropriate based on first selection. See guideline on enabled/disabled selectable fields. 
"Run" should have a tooltip that says: Validate input agains lineage

Screenshot 2023-05-02 at 5 12 01 pm

### LINKED ISSUE #35: Example selection repositioning
Reposition the Example block to the right, alongside the dark/light mode toggle.
Add a label before the example selection box "Select your example" <BOX> " or paste your own code" 
The box should follow the design kit guidelines

The basic (most simple) example should be selected

Screenshot 2023-05-02 at 5 21 24 pm

### PR DESCRIPTION
**Pull request by joanlopez**
**What this PR does / why we need it**:

It replaces the current combination of `<button>` + `<input>` for the "Share" feature, by an `<IconButton>` and a `<Modal>` from `@grafana/ui`.

It also introduces the use of `GrafanaTheme2` from `@grafana/data` as the baseline for the transition described at #25.

Finally, it adds an initial version of a `Makefile` to make the contributor's life better.

**Which issue(s) this PR fixes**:

Fixes #25 
Fixes #33 
Fixes #34 
Fixes #35 

**Special notes for your reviewer**:

Not 100% sure about some details, like the copies or the new position , 

Some code comments inline. 

**Screenshots**:

image

Captura de pantalla 2023-05-05 a las 11 45 58

### CODE REVIEW

File: src/components/Header/Share.tsx [hunk 1]
```diff
@@ -37,32 +24,63 @@ const Share = () => {
...
+
+  const styles = useStyles2(getStyles);
+
   return (
     <>
-      <button style={styles.btn} onClick={shareFn}>
-        Share
-      </button>
-      <input style={styles.input} ref={shareRef} onClick={focusFn} readOnly={true} value={shareUrl(shareId || '')} />
+      <div className={styles.iconContainer}>
```
Comment (joanlopez) [inline src/components/Header/Share.tsx:60]: I tried to set the `className` directly to the `IconButton`, but it gets overwritten by its default styles.

File: src/index.tsx [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
-import { ThemeProvider } from './theme';
+import { ThemeContext, ThemeProvider } from './theme';
 import { StateProvider } from './state';
 import { getWebInstrumentations, initializeFaro } from '@grafana/faro-web-sdk';
+import { ThemeContext as GrafanaThemeContext } from '@grafana/ui';
 import { TracingInstrumentation } from '@grafana/faro-web-tracing';
 import { ReactIntegration } from '@grafana/faro-react';
+import React, { useContext } from 'react';
+
+const GrafanaTheme = (props: React.PropsWithChildren) => {
```
Comment (joanlopez) [inline src/index.tsx:12]: Temporal workaround until we finish #25 and get rid of the custom `ThemeProvider` in favor of `ThemeContext` from `@grafana/ui`.

File: src/theme.tsx [hunk 3]
```diff
@@ -8,23 +9,27 @@ export enum Theme {
 interface ThemeCfg {
   theme: Theme;
   toggleTheme: React.DispatchWithoutAction;
+  grafanaTheme: GrafanaTheme2;
```
Comment (joanlopez) [inline src/theme.tsx:12]: Same as https://github.com/grafana/play-thema/pull/39/files#r1185973789

**Comment by undef1nd** [comment]
I think `Copy` button may need to be moved to be consistent with this UI, for example:
Screenshot 2023-05-05 at 13 48 08

**Comment by Clarity-89** [comment]
Looks good from the UI point of view, however there are some issues here that we need to think about:

Before we introduce changes like this it's important to remember that most of the Grafana packages are meant to be run in Grafana itself and are not guaranteed to work properly when run standalone. Since the thema playground is not run inside Grafana, there will be some limitations. I'm quite surprised this PR works at all, as grafana/ui is one of those packages that definitely need Grafana to work properly. There might've been some changes there, so I'll need to investigate. 

Staring from https://github.com/grafana/play-thema/pull/26, I'm getting a bunch of warnings in the terminal about missing source maps (probably related to the added @grafana/toolkit) and cannot run any WASM functions (`TypeError: Failed to execute 'compile' on 'WebAssembly': Incorrect response MIME type. Expected 'application/wasm'.`) Does it work for everyone? I see this seems to work in the deployed version so maybe it's only happening for me. 

About the changes themselves, I think the modal is a bit redundant here, we can just add a copy button that would copy the URL on click, as is done [here](https://www.typescriptlang.org/play) for example. 

I should have probably mentioned the limitations of using Grafana packages outside of Grafana earlier, sorry for that!

**Comment by Clarity-89** [comment]
> So far, I think we just need choose what to do? For instance, is it fine to use components from `@grafana/ui` while they work correctly? 

I checked with the plugins platform team and it'd be fine to use it as long as we don't use any Grafana-dependent features (e.g. visualisations), so I think we good there. Just need to upgrade grafana/ui to the latest (9.5.1) version.

> From the user point of view, I'd agree, I prefer the more minimalist, modal-free, approach. Indeed that's the current behavior if I'm not wrong haha 😅

Let's then keep the current behavior and add the modal later if it's really necessary. I can take over this PR and update the grafana/ui usage, so you can concentrate on the thema-specific Go things :)

**Comment by Clarity-89** [comment]
@joanlopez sorry to hijack your PR, but after adding grafana/ui I realised that there are many changes needed to make the theme work 😅 

Now the Playground uses the components from grafana/ui. There are still a few fixes to be done, but they can wait for a separate PR.

**Screenshots:**
Before:
Screenshot 2023-05-12 at 18 35 26

After: 

Screenshot 2023-05-12 at 18 36 39

**Review by joanlopez — COMMENTED** [review]
I'm not sure which of these is due to the changes present on this pull request, and which ones were not working properly before that, or maybe some are due to my development environment, but here's a list of what I found not working well while testing the PR's branch:

- The `version` dropdown should be disabled once the user selects an operation where no version needs to be specified (according to #35).

- The experience of updating the contents of the code editor containing the lineage is annoying because there's considerable delay after every modification (as if it'd be doing some blocking work after key click) - plus no feedback at all, which makes the whole experience even more confusing.

Other than that, the code now looks much better and the UI more consistent, so after these few minor adjustments, I think we'll be ready to merge it! Awesome work @Clarity-89, thanks!

**Comment by joanlopez** [comment]
Also, worth noting this will:
- Fix #25 
- Contribute #33 #34 #35 

🎉 🎉 🎉 

PS: I'm not sure about which of these will we be able to consider as finished, cause there's some open design questions. Design wise, I'm happy with the result of this pull request, so I'd consider them as done!

**Comment by Clarity-89** [comment]
@joanlopez thanks for the review! I had to split the app context and break down some parts into smaller components to prevent excessive number of renders and the code editors should work properly now. Also fixed the version dropdown. Let me know if you find any other issues.

**Comment by undef1nd** [comment]
The UI looks way better now! 🧡 Thank you!

**Pull request merged**